### PR TITLE
[WIP] Referral code implementation in App

### PR DESCRIPTION
**Pull request by avikumar15**

### CODE REVIEW

File: app/src/main/java/org/pragyan/dalal18/ui/RegistrationActivity.kt [hunk 1]
```diff
@@ -86,6 +86,7 @@ class RegistrationActivity : AppCompatActivity() {
                             .setFullName(nameEditText.text.toString())
                             .setPassword(passwordEditText.text.toString())
                             .setUserName(nameEditText.text.toString())
+                            .setReferralCode(referralCodeEditText.text.toString())
```
Comment (ranaaditya) [inline app/src/main/java/org/pragyan/dalal18/ui/RegistrationActivity.kt:89]: Need to add **InvalidReferralCodeError** also !

refer [this](https://github.com/delta/dalal-street-proto/blob/master/actions/Register.proto) file

**Review by ranaaditya — CHANGES_REQUESTED** [review]

File: app/src/main/java/org/pragyan/dalal18/ui/RegistrationActivity.kt (hunk 1 above)
Comment (avikumar15) [reply app/src/main/java/org/pragyan/dalal18/ui/RegistrationActivity.kt:89]: There are only 6 fields in Register Request proto file you linked above.
```
    string email = 1;
    string password = 2;
    string user_name = 3;
    string full_name = 4;
    string country = 5;
    string referral_code = 6;
```

Where is InvalidReferralCodeError? Check the generated .java file from this proto, there isn't any such field there either.

Comment (Harsh2098) [inline app/src/main/java/org/pragyan/dalal18/ui/RegistrationActivity.kt:89]: Shouldn't you be checking for `InvalidReferralCodeError` below-

`val message = when (response.statusCode) {`

And then depending on your flow, probably not take him back to login page.

(PS why am I even reviewing this xD)

Comment (ranaaditya) [reply app/src/main/java/org/pragyan/dalal18/ui/RegistrationActivity.kt:89]: > There are only 6 fields in Register Request proto file you linked above.
> 
> ```
>     string email = 1;
>     string password = 2;
>     string user_name = 3;
>     string full_name = 4;
>     string country = 5;
>     string referral_code = 6;
> ```
> 
> Where is InvalidReferralCodeError? Check the generated .java file from this proto, there isn't any such field there either.

not here, below this code where we have several response messages !

Comment (avikumar15) [reply app/src/main/java/org/pragyan/dalal18/ui/RegistrationActivity.kt:89]: > Shouldn't you be checking for `InvalidReferralCodeError` below-
> 
> `val message = when (response.statusCode) {`
> 
> And then depending on your flow, probably not take him back to login page.
> 
> (PS why am I even reviewing this xD)

Oh yeah, the response code. Got it xD

Comment (avikumar15) [reply app/src/main/java/org/pragyan/dalal18/ui/RegistrationActivity.kt:89]: Yeah got it, you pointed it at the request builder so got confused.

**Comment by ranaaditya** [comment]
Commits of this PR has been merged in [this](https://github.com/delta/dalal-street-android/pull/180) PR.

Thanks @avikumar15 for this wonderful contribution :partying_face:

**Pull request closed without merging**